browser/src/neovim/NeovimInstance.ts

```diff
@@ -534,6 +543,45 @@ export class NeovimInstance extends EventEmitter implements INeovimInstance {
         return path.normalize(currentWorkingDirectory)
     }
 
+    public async getVimHighlights(): Promise<ITokenColorsSetting[]> {
+        const highlights = [
+            "identifier",
+            "function",
+            "constant",
+            "preproc",
+            "type",
+            "foldbraces",
+            "define",
+            "keyword",
+            "statement",
+            "comment",
+            "normal",
+        ]
+
+        const colorValues = await Promise.all(
+            highlights.map(async token => {
+                const highlight = (await this._neovim.request("nvim_get_hl_by_name", [
+                    token,
+                    true,
+                ])) as IHighlight
+                const highlightOrDefault = (color: number) => (color ? Color(color).hex() : null)
+                const { bold, foreground, background, italic } = highlight
+
+                return {
+                    scope: [token],
+                    settings: {
+                        fallback: token,
```

Comment:
The fact that the system end-to-end still needs to know about the Vim highlight group means we haven't fully decoupled it yet. It'd be great if we could break our dependency on this, because it adds complexity along the entire stack. If we can just speak in `tokenColors` without having to worry about resolving a fallback, that would make life easier everywhere we use this.

One idea we could try is to _invert_ that responsibility. Right now, we create a bunch of `tokenColors` and specify a `fallback`. What if, instead, we create a set of default `editor.tokenColors` from this method?

For example, in `QuickInfoContainer`, we have this list of scopes:
```
...
       "variable.language": {
            scope: [],
            settings: {
                fallback: "Identifier",
            },
        },
        "variable.parameter": {
            scope: [],
            settings: {
                fallback: "Identifier",
            },
        },
```

Could we instead, here, specify a mapping like:
```
const VimHighlightGroupsToScope = {
"Identifier": ["variable.language", "variable.parameter"]
}

And then, from this method, return essentially that full set?
```

We're basically inverting the dependency, and letting `NeovimInstance` do the job of converting completely to the `ITokenColor` world. Then, if this is the default set, they can naturally be overridden by the theme and user configuration. 

This gives us a few benefits:
- Consumers don't need to worry about resolving `fallback` or what `fallback` means
- We don't have to reimplement the logic in `QuickInfoContainer` in different places

I would prefer to not have the vim highlight group as part of the `ITokenColor` model (even as `fallback`), because of the coupling it creates for the consumer - the consumer has to know about vim highlight groups to work. It seems like we can push that logic into `getVimHighlights` by inverting how we create our token colors - letting that function give us a full set, instead of the consumer needing to reverse map.